ibazel/command/notify_command_test.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2017 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package command
+
+import (
+	"errors"
+	"os/exec"
+	"syscall"
+	"testing"
+
+	"github.com/bazelbuild/bazel-watcher/bazel"
+	mock_bazel "github.com/bazelbuild/bazel-watcher/bazel/testing"
+)
+
+func TestNotifyCommand(t *testing.T) {
+	cmd := exec.Command("cat")
+	cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
+
+	c := &notifyCommand{
+		args:      []string{"moo"},
+		bazelArgs: []string{},
+		cmd:       cmd,
+		target:    "//path/to:target",
+	}
+
+	if c.IsSubprocessRunning() {
+		t.Errorf("New subprocess shouldn't have been started yet. State: %v", cmd.ProcessState)
+	}
+
+	var err error
+	c.stdin, err = cmd.StdinPipe()
+	if err != nil {
+		t.Error(err)
+	}
+
+	// Mock out bazel to return non-error on test
+	b := &mock_bazel.MockBazel{}
+	b.WaitError(nil)
+	bazelNew = func() bazel.Bazel { return b }
+	defer func() { bazelNew = oldBazelNew }()
+
+	c.NotifyOfChanges()
+	b.WaitError(errors.New("Demo error"))
+	c.NotifyOfChanges()
+	b.WaitError(nil)
+	c.NotifyOfChanges()
+
```

Comment:
Might be good to expect that isSubprocessRunning to be true at this point

Reply:
Done

Reply:
Huh, turns out that isn't possible to do because I'm using `cmd.CombinedOutput` which starts the job. I'm going to return it to the old way and work instead on an end to end test which is a better test anyway.